modules/context/auth.go

```diff
@@ -31,10 +32,23 @@ func Toggle(options *ToggleOptions) macaron.Handler {
 		}
 
 		// Check prohibit login users.
-		if ctx.IsSigned && ctx.User.ProhibitLogin {
-			ctx.Data["Title"] = ctx.Tr("auth.prohibit_login")
-			ctx.HTML(200, "user/auth/prohibit_login")
-			return
+		if ctx.IsSigned {
+
+			if ctx.User.ProhibitLogin {
+				ctx.Data["Title"] = ctx.Tr("auth.prohibit_login")
+				ctx.HTML(200, "user/auth/prohibit_login")
+				return
+			}
+
+			// Prevent infinite redirection loop
+			if ctx.Req.URL.Path == "/user/change_password" {
```

Comment:
This will not work if appsuburl is used

Reply:
@lafriks I don't quite get this ... By the way, I fixed the other review with https://github.com/go-gitea/gitea/pull/4489/commits/586c0802a3f596a62a8da3877240cdf33881f8d8

Reply:
@adelowo if gitea is hosted as `https://example.com/gitea/` as root url than url path would be `/gitea/user/change_password`